hbase-server/src/test/java/org/apache/hadoop/hbase/master/http/TestMasterHbckServlet.java

```diff
@@ -0,0 +1,237 @@
+  static final Path FAKE_PATH = new Path(
+    "/hbase/data/default/" + FAKE_TABLE.getTableName() + "/" + FAKE_HRI_3.getEncodedName());
+  static final long FAKE_START_TIMESTAMP = System.currentTimeMillis();
+  static final long FAKE_END_TIMESTAMP = System.currentTimeMillis() + 1000;
+  static final Gson GSON = GsonFactory.buildGson();
+
+  private HMaster master;
+
+  @Before
+  public void setupMocks() {
+    // Fake inconsistentRegions
+    Map<String, Pair<ServerName, List<ServerName>>> inconsistentRegions = new HashMap<>();
+    inconsistentRegions.put(FAKE_HRI.getEncodedName(),
+      new Pair<>(FAKE_HOST, Arrays.asList(FAKE_HOST_2)));
+
+    // Fake orphanRegionsOnRS
+    Map<String, ServerName> orphanRegionsOnRS = new HashMap<>();
+
+    // Fake orphanRegionsOnFS
+    Map<String, Path> orphanRegionsOnFS = new HashMap<>();
+    orphanRegionsOnFS.put(FAKE_HRI_3.getEncodedName(), FAKE_PATH);
+
+    // Fake HbckChore
+    HbckChore hbckChore = mock(HbckChore.class);
+    doReturn(FAKE_START_TIMESTAMP).when(hbckChore).getCheckingStartTimestamp();
+    doReturn(FAKE_END_TIMESTAMP).when(hbckChore).getCheckingEndTimestamp();
+    doReturn(inconsistentRegions).when(hbckChore).getInconsistentRegions();
+    doReturn(orphanRegionsOnRS).when(hbckChore).getOrphanRegionsOnRS();
+    doReturn(orphanRegionsOnFS).when(hbckChore).getOrphanRegionsOnFS();
+
+    // Fake region holes
+    List<Pair<RegionInfo, RegionInfo>> holes = new ArrayList<>();
+
+    // Fake region overlaps
+    List<Pair<RegionInfo, RegionInfo>> overlaps = new ArrayList<>();
+    overlaps.add(new Pair<>(FAKE_HRI, FAKE_HRI_2));
+
+    // Fake unknown servers
+    List<Pair<RegionInfo, ServerName>> unknownServers = new ArrayList<>();
+    unknownServers.add(new Pair<>(FAKE_HRI, FAKE_HOST_2));
+
+    // Fake empty region info
+    List<String> emptyRegionInfo = new ArrayList<>();
+    emptyRegionInfo.add(FAKE_HRI_3.getEncodedName());
+
+    // Fake catalog janitor report
+    Report report = mock(Report.class);
+    doReturn(FAKE_START_TIMESTAMP).when(report).getCreateTime();
+    doReturn(holes).when(report).getHoles();
+    doReturn(overlaps).when(report).getOverlaps();
+    doReturn(unknownServers).when(report).getUnknownServers();
+    doReturn(emptyRegionInfo).when(report).getEmptyRegionInfo();
+
+    // Fake CatalogJanitor
+    CatalogJanitor janitor = mock(CatalogJanitor.class);
+    doReturn(report).when(janitor).getLastReport();
+
+    // Fake master
+
+    master = mock(HMaster.class);
+    doReturn(HBaseConfiguration.create()).when(master).getConfiguration();
+    doReturn(true).when(master).isInitialized();
+    doReturn(Optional.of(FAKE_HOST)).when(master).getActiveMaster();
+    doReturn(janitor).when(master).getCatalogJanitor();
+    doReturn(hbckChore).when(master).getHbckChore();
+    MasterRpcServices services = mock(MasterRpcServices.class);
+    doReturn(services).when(master).getRpcServices();
+    doReturn(services).when(master).getMasterRpcServices();
```

Comment:
Why do all this mockery? At this point, it's it simpler to stand up a mini-cluster and manufacture the state you need? It looks like `TestMetaFixer` has some utility code for manufacturing faulty states.

Reply:
Because when I wrote this test this was the way I understood it could be implemented.